staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -597,58 +599,85 @@ func fieldErrorForReason(fldPath *field.Path, value interface{}, detail string,
 
 // evalMessageExpression evaluates the given message expression and returns the evaluated string form and the remaining budget, or an error if one
 // occurred during evaluation.
-func evalMessageExpression(ctx context.Context, expr celgo.Program, exprSrc string, activation interpreter.Activation, remainingBudget int64) (string, int64, *cel.Error) {
+func evalMessageExpression(ctx context.Context, expr celgo.Program, exprSrc string, activation interpreter.Activation, remainingBudget int64) ([]string, int64, *cel.Error) {
```

Comment:
Do we also need to pass in a `allowsMultipleMessageExpressions` bool to this function to reject list responses for VAP since we don't type check the expressions and hence don't know for sure what the response type will be until runtime?

Reply:
This function is only called from the CRD Validation Rules CEL Validator, so that may not be necessary.

Reply:
Are we also adding mutliple message expression support to VAP or other places CEL is supported?  Should we?

Reply:
We can do that for parity. I can follow-up later on. It would be important to do this release if we want to use the same CEL library trick, I think. Otherwise we'd need a separate flag for VAP